KLR/Trace/NKI.lean

```diff
@@ -104,6 +104,25 @@ private def lookupName (name : Name) : Trace Term := do
     | .str n id => (<- lookupName n).attr id
     | _ => throw s!"{name} not found"
 
+/-
+Best effort checks for slice overflow
+
+Currently this implementation will silently clip out-of-bounds slice patterns,
+similar to Python and numpy. However, to aide developers we will (try to) warn
+or fail if we detect an overflow. This function is applied to function
+arguments.
+-/
+private def checkAccess (t : Term) (warnOnly : Bool := true) : Trace Unit := do
```

Comment:
All this uses is `b`. I wonder if we should be checking it earlier, like when you make the `b` in the first place.